src/js/components/ClusterDropdown.js

```diff
@@ -0,0 +1,172 @@
+import * as React from "react";
+import mixin from "reactjs-mixin";
+import { Dropdown } from "reactjs-components";
+import { StoreMixin } from "mesosphere-shared-reactjs";
+import { MountService } from "foundation-ui";
+import { routerShape } from "react-router";
+
+import MesosSummaryStore from "#SRC/js/stores/MesosSummaryStore";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+import SidebarActions from "#SRC/js/events/SidebarActions";
+import ClipboardTrigger from "#SRC/js/components/ClipboardTrigger";
+
+function ClusterDropdownTrigger({ onTrigger, children }) {
+  return (
+    <span className="header-bar-dropdown-trigger" onClick={onTrigger}>
+      {children}
+    </span>
+  );
+}
+const METHODS_TO_BIND = ["handleItemSelect", "handleTextCopy"];
+export default class ClusterDropdown extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      isTextCopied: false
+    };
+
+    this.store_listeners = [
+      {
+        name: "metadata",
+        events: ["success"],
+        listenAlways: false
+      },
+      {
+        name: "summary",
+        events: ["success"],
+        listenAlways: false
+      }
+    ];
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getClusterName() {
+    const states = MesosSummaryStore.get("states");
+
+    return states ? states.getClusterName() : null;
```

Comment:
so this can return `null` if the clustername wasnt fetched yet, correct? 
dont rellay like it TBH :( (and this is your code, you added `getClusterName` to SummaryList :D 

Maybe some generic `Cluster Options` text would be better 🤔 

...wasnt able to "break" it while testing tho -> not blocking

Reply:
I understand your concerns and I think we should make it a lot better when moving to the data layer. I also don't see it as blocking, therefore I am merging 😇